helmfile/charts/user-rbac/templates/clusterroles/user-crds.yaml

```diff
@@ -0,0 +1,10 @@
+{{- if .Values.userCRDs.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: user-crds
+rules:
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
```

Comment:
Could we use [`resourceNames`](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#:~:text=resourceNames)? If not, can you please leave a comment explaining to the next person why we cannot use resourceNames and reassure them that OPA will deny requests of unsuitable CRDs?

Reply:
I know I tried resourceNames for something else and it did not work. So i dropped that thought. But now I look at this, yeah it should be possible. I will try it out. 
But if that works then the OPA policy maybe is not needed? :cry:

One reason that opa can be better, is that it can check on multiple variables instead of just the name. 
At the moment the name and group need to match.

Reply:
Okay, I found why resourceNames does not work here. It is not possible to restrict create or delete with resourceNames. So we must rely on OPA for this.

```
Note: You cannot restrict create or deletecollection requests by their resource name. For create, this limitation is because the name of the new object may not be known at authorization time.
```

Reply:
Mkey, and I guess pre-creating the CRD or not allowing its deletion would lead to poor DX, right? If so, please just add a comment.

Reply:
Pre-creating and preventing deletion it would lead to inability to follow normal "here's how you add X to your cluster" instructions, so if we can enable people to just follow normal "helm install"-types of guides, it would be very nice from a DX perspective.